src/udev/udev-node.c

```diff
@@ -70,18 +72,18 @@ static int node_symlink(sd_device *dev, const char *devnode, const char *slink)
         if (r < 0)
                 return log_device_debug_errno(dev, r, "Failed to get relative path from '%s' to '%s': %m", slink, devnode);
 
-        r = device_get_device_id(dev, &id);
+        r = tempfn_xxxxxx(slink, NULL, &slink_tmp);
         if (r < 0)
-                return log_device_debug_errno(dev, r, "Failed to get device id: %m");
-
-        slink_tmp = strjoina(slink, ".tmp-", id);
-        (void) unlink(slink_tmp);
+                log_device_debug_errno(dev, r,
+                                       "Failed to generate temporary file name for symlink '%s', "
+                                       "creating symlink non-atomically, ignoring: %m",
+                                       slink);
```

Comment:
why make this graceful? why not just fail this? is there any valid case where tempfn_xxxxx() can fail here?